File: 엄마호랑이반/19주차 문제/이다영/서버증설.java
import java.util.*;

class Solution {
    public int solution(int[] players, int m, int k) {
        int cnt = 0;
        int[] servers = new int[24];
        for (int t = 0; t < 24; t++) {
            int player = players[t];
            int server = servers[t];
            
            if (player < m || player < (server*m) || (server*m <= player && player < (server+1)*m)) {
                continue;
            }
            
            int x = (player - (server*m)) / m;
            cnt += x;
            for (int i = 0; i < k; i++) {
                if (t+i > 23) break;
                servers[t + i] += x;
            }
        }
        return cnt;
    }
}

File: 엄마호랑이반/19주차 문제/박정환/서버 증설.py
def solution(players, m, k):
    answer = 0
    server = [0] * len(players)

    for i in range(24):

        if (players[i] // m) > server[i]:
            tmp = (players[i] // m) - server[i]
            answer+= tmp
            if i + k < 24:
                for j in range(i, i + k):
                    server[j] += tmp
            else:
                for j in range(i, 24):
                    server[j] += tmp

    return answer


print(solution([0, 2, 3, 3, 1, 2, 0, 0, 0, 0, 4, 2, 0, 6, 0, 4, 2, 13, 3, 5, 10, 0, 1, 5], 3, 5))
